src/configobj/__init__.py

```diff
@@ -757,7 +758,7 @@ def merge(self, indict, decoupled=False):
         """
         for key, val in list(indict.items()):
             if decoupled:
-                val = copy.deepcopy(val)
+                val = copy(val)
```

Comment:
There impact is quite unhidden, you defeat the purpose of the ``decoupled`` flag. Also problematic: changing this for a single regex is unwise, since (mostly) _all_ of them need to become dynamic for implementing in-line comments with alternate markers. A way better solution would be to put them all into a utility class (taking the necessary params to build the regex), that can then be aggregated and used by ConfigObj.